circus-train-gcp/src/main/java/com/hotels/bdp/circustrain/gcp/HdfsGsCredentialAbsolutePathFactory.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.circustrain.gcp;
+
+import org.apache.hadoop.fs.Path;
+
+public class HdfsGsCredentialAbsolutePathFactory {
+
+  Path newInstance(Path directory, String randomString) {
```

Comment:
Since you're making the dir random, you don't need to make the file random as well, so the random string can go, which makes this class trivial and it can go altogether, just inline it in the copier class. One factory less.